app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/BeaconViewFragment.java

```diff
@@ -147,8 +148,11 @@ protected void onColoringModeSelected(@ColorUtil.ColoringMode int coloringMode,
         for (Beacon beacon : beaconManager.getBeaconMap().values()) {
             for (BeaconFilter beaconFilter : beaconFilters) {
                 if (beaconFilter.matches(beacon)) {
-                    beacons.add(beacon);
-                    break;
+                    //TODO remove beacon filter
+                    if (Objects.equals(beacon.getMacAddress(), "E2:38:2E:68:46:E9")) {
```

Comment:
You could have adjusted the filter, that's what it's for. Also, to compare strings, use `beacon.getMacAddress().equals("E2:38:2E:68:46:E9")`